src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -1005,6 +1054,107 @@ void Graph::PullOutputData(BlobMap &out) {
     }
 }
 
+void Graph::InferStatic(InferRequestBase* request) {
+    dnnl::stream stream(eng);
+
+    for (const auto& node : executableGraphNodes) {
+        VERBOSE(node, config.verbose);
+        PERF(node, config.collectPerfCounters);
+
+        if (request)
+            request->ThrowIfCanceled();
+        ExecuteNode(node, stream);
+    }
+}
+
+void Graph::InferDynamic(InferRequestBase* request) {
+    dnnl::stream stream(eng);
+
+    std::set<size_t> syncIndsWorkSet;
+    for (const auto& nodeIndx : syncNodesInds) {
+        syncIndsWorkSet.insert(nodeIndx.second);
+        //since sometimes we need to run the synchronization node  alone (for example in the case of internal dynamism)
+        //let's add another sync index after the sync point node
+        syncIndsWorkSet.insert(nodeIndx.second + 1);
```

Comment:
It looks like in general we should distinguish between output shape dependency on inputs and real internal dynamism. Should decrease number of sync points. Don't know will it significantly improve the perf though.

Reply:
Local performance tests revealed that running the barrier nodes separately from the parallel shapeInfer/prepare params sequence, does not have noticeable impact on  the overall performance.